xamarin/Xamarin.Forms
# Uwp ui tests

### Description of Change ###

Setup yaml/cake tasks to automate UWP tests

I fixed a bunch of warnings as well that were causing errors when building the CG directly. 

### API Changes ###

#### Cake Tasks
- cg-uwp: builds CG UWP and generates appx files against temporary PFX file
- cg-uwp-build-tests: builds UWP UI Tests
- cg-uwp-deploy: deploy: appx packages generated from cg-uwp
- cg-uwp-run-tests: run nunit test dll
- cg-uwp-run-tests-ci: deploy and run nunit tests for release builds
 

### Testing Procedure ###
- make sure all the ui test builds still work
- make sure UWP tests run

### PR Checklist ###
